Plugins: Add indexed plugin selectors (#72742)
Automattic/wp-calypso: client/state/plugins/installed/README.md

Get the most recent status for a plugin action (including "inProgress" for currently-running actions).

### `getAllPluginsIndexedByPluginSlug( state: Object )`

> Get an object for retrieving a plugin by its slug

Reviewer: ```suggestion
Get an object of all installed plugins in all sites, indexed by the plugin slug.
```

### `getAllPluginsIndexedBySiteId( state: Object )`

> Get an object for retrieving the plugins for a site by a site ID

Reviewer: This can be a bit confusing since we don't really pass a site ID, rather they're indexed by it, and it's not retrieving the plugins of a site, but rather all plugins indexed by site. Also, is this really about retrieving or not? 

Should we change it to something like:

```suggestion
Get an object of all installed plugins of each site, indexed by site ID.
```
